MAVSDK Windows Installation

Hi, 
When installing MAVSDK on Windows, are we given the option to either 1) locally install mavsdk-windows-x64-release.zip or 2) clone the repository and cmake build and install? The zip folder does not contain a CMakeLists.txt so do we have to make one ourselves? Because of this uncertainty I tried the second approach by using:
git clone https://github.com/mavlink/MAVSDK.git
git submodule update --init --recursive
cmake -DCMAKE_BUILD_TYPE=Debug -Bbuild/default -S. 
cmake --build build/default -j8 
And was given this error message on the last command:
![Capture1](https://user-images.githubusercontent.com/59535719/150206077-e8e5396a-bf28-4e2e-98b5-df698c934ab8.PNG)
Note the message is a lot longer but the errors all involve unresolved external symbols.

I see the same on my local machine:

```
`telemetry_health.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum
mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,clas
s std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::
callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@s
td@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcx
proj]
telemetry_sync.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum ma
vsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class
std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::ca
llback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std
@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxpr
oj]
mavlink_passthrough.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(en
um mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,c
lass std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetaile
d::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@
D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.
vcxproj]
mission_transfer_lossy.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl
(enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const
&,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDeta
iled::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_trai
ts@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runn
er.vcxproj]
system_connection_async.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdec
l(enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const
 &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDet
ailed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_tra
its@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_run
ner.vcxproj]
statustext.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk
::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std:
:basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callba
ck" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?
$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
telemetry_async.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum m
avsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class
 std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::c
allback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@st
d@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxp
roj]
telemetry_gps_origin.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(e
num mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,
class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetail
ed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits
@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner
.vcxproj]
offboard_acceleration.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(
enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &
,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetai
led::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_trait
s@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runne
r.vcxproj]
offboard_velocity.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum
 mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,cla
ss std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed:
:callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@
std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vc
xproj]
offboard_position.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum
 mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,cla
ss std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed:
:callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@
std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vc
xproj]
param.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk::log
::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basi
c_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callback" (
?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allo
cator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
mission_rtl.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsd
k::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std
::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callb
ack" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V
?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
mission_raw_mission_changed.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __
cdecl(enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > c
onst &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::Lo
gDetailed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char
_traits@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests
_runner.vcxproj]
mission_raw_import_and_fly.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __c
decl(enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > co
nst &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::Log
Detailed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_
traits@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_
runner.vcxproj]
mission_set_current.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(en
um mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,c
lass std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetaile
d::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@
D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.
vcxproj]
log_files.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk:
:log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::
basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callbac
k" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$
allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
mission_cancellation.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(e
num mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,
class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetail
ed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits
@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner
.vcxproj]
mission_change_speed.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(e
num mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,
class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetail
ed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits
@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner
.vcxproj]
mission.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk::l
og::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::ba
sic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callback"
 (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$al
locator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
follow_me.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk:
:log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::
basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callbac
k" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$
allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
geofence_inclusion.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enu
m mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,cl
ass std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed
::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D
@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.v
cxproj]
gimbal.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk::lo
g::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::bas
ic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callback"
(?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$all
ocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
info.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsdk::log:
:Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic
_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callback" (?
callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$alloc
ator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
camera_mode.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsd
k::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std
::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callb
ack" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V
?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
camera_settings.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum m
avsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class
 std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::c
allback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@st
d@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxp
roj]
camera_status.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mav
sdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class s
td::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::cal
lback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@
@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxpro
j]
camera_take_photo.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum
 mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,cla
ss std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed:
:callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@
std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vc
xproj]
action_goto.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsd
k::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std
::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callb
ack" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V
?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
action_hold.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsd
k::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std
::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callb
ack" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V
?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
calibration.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum mavsd
k::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std
::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed::callb
ack" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@std@@V
?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
camera_capture_info.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(en
um mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,c
lass std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetaile
d::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@
D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.
vcxproj]
action_hover_async.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enu
m mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,cl
ass std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed
::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D
@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.v
cxproj]
action_hover_sync.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdecl(enum
 mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,cla
ss std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDetailed:
:callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_traits@D@
std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vc
xproj]
action_takeoff_and_kill.obj : error LNK2001: unresolved external symbol "public: static class std::function<bool __cdec
l(enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const
 &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)> mavsdk::LogDet
ailed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_string@DU?$char_tra
its@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_run
ner.vcxproj]
action_transition_multicopter_fixedwing.obj : error LNK2001: unresolved external symbol "public: static class std::func
tion<bool __cdecl(enum mavsdk::log::Level,class std::basic_string<char,struct std::char_traits<char>,class std::allocat
or<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,int)
> mavsdk::LogDetailed::callback" (?callback@LogDetailed@mavsdk@@2V?$function@$$A6A_NW4Level@log@mavsdk@@AEBV?$basic_str
ing@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@1H@Z@std@@A) [C:\src\MAVSDK\build\default\src\integration_tests\integ
ration_tests_runner.vcxproj]
C:\src\MAVSDK\build\default\src\integration_tests\Debug\integration_tests_runner.exe : fatal error LNK1120: 1 unresolve
d externals [C:\src\MAVSDK\build\default\src\integration_tests\integration_tests_runner.vcxproj]
```

--- a/src/mavsdk/core/include/mavsdk/log_callback.h
+++ b/src/mavsdk/core/include/mavsdk/log_callback.h
@@ -14,6 +14,7 @@ enum class Level : int { Debug = 0, Info = 1, Warn = 2, Err = 3 };
 using Callback =
     std::function<bool(Level level, const std::string& message, const std::string& file, int line)>;
 
-void subscribe(const Callback& callback);
+extern Callback& get_callback();
+extern void subscribe(const Callback& callback);
 
 } // namespace mavsdk::log

--- a/src/mavsdk/core/log.cpp
+++ b/src/mavsdk/core/log.cpp
@@ -20,11 +20,16 @@
 
 namespace mavsdk {
 
-log::Callback LogDetailed::callback = nullptr;
+static log::Callback callback_{nullptr};
+
+log::Callback& log::get_callback()
+{
+    return callback_;
+}
 
 void log::subscribe(const log::Callback& callback)
 {
-    LogDetailed::callback = callback;
+    callback_ = callback;
 }
 
 void set_color(Color color)

--- a/src/mavsdk/core/log.h
+++ b/src/mavsdk/core/log.h
@@ -47,7 +47,8 @@ public:
 
     virtual ~LogDetailed()
     {
-        if (callback && callback(_log_level, _s.str(), _caller_filename, _caller_filenumber)) {
+        if (log::get_callback() &&
+            log::get_callback()(_log_level, _s.str(), _caller_filename, _caller_filenumber)) {
             return;
         }
 
@@ -121,8 +122,6 @@ public:
 
     LogDetailed(const mavsdk::LogDetailed&) = delete;
     void operator=(const mavsdk::LogDetailed&) = delete;
-
-    static log::Callback callback;
 
 protected:
     log::Level _log_level = log::Level::Debug;

